src/ERC20Rebasing.sol

```diff
@@ -23,11 +23,12 @@ abstract contract ERC20Rebasing is ERC20 {
     function balancePerShare() public view virtual returns (uint128);
 
     function sharesToBalance(uint256 shares) public view returns (uint256) {
-        return mulDiv18(shares, balancePerShare()); // floor
+        // WAD = 1e18
+        return FixedPointMathLib.fullMulDiv(shares, balancePerShare(), 1e18); // floor
```

Comment:
Does mulWad work here or overflow?

Reply:
Overflow

Reply:
Then the other use of mulWad should be changed to fullMulDiv